frontend/src/components/modals/settings/SettingsForm.tsx

```diff
@@ -52,13 +52,19 @@ function SettingsForm({
   const [enableAdvanced, setEnableAdvanced] =
     React.useState(advancedAlreadyInUse);
 
+  const handleAdvancedChange = (value: boolean) => {
+    setEnableAdvanced(value);
+    // Reset the base URL if the user disables advanced options
+    if (!value) onBaseURLChange("");
```

Comment:
Should we also clear security analyzer etc?

Reply:
That depends on what behaviour we want. What if the user is using a model we provide AND the security analyzer? If we clear when the advanced option is false, then we are only allowing the security analyzer, confirmation mode, etc... with custom models.

Right now the user can enable confirmation mode, switch to another model, and the mode is still enabled. We probably don't want the same behaviour for base URL though since that is (AFAIK) directly tied to the custom model, so it makes sense to clear it when advanced options are off.

IMO we should find better UI/UX for this, especially if settings will expand

Reply:
Just to note, we can use the Advanced Settings window for "standard" models too, not only custom models. It may be a little more to type, but I'm pretty sure it works.